nixos/modules/services/logging/beats/lib.nix

```diff
@@ -0,0 +1,145 @@
+{ lib, pkgs, ... }:
+
+with lib;
+
+{
+
+  mkCommonOptions = { name, defaults }: {
```

Comment:
Unless I'm missing something here, you don't need to abstract this with a library file. Just write an option that you can set multiple times wherever you need it. Something like

```nix
{
  options.services.beats = mkOption {
    type = types.attrsOf (types.submodule ({ name, ... }: {
      options = {
        enable = true;
        # ...
      };
    }));
  };
}
```

And then set `services.beats.apm-server.enable = true;` and such where you need an instance. This way you also won't duplicate all these options.